backend/authors/views.py

```diff
@@ -0,0 +1,70 @@
+from django.shortcuts import render
+from django.http import HttpResponse, JsonResponse
+from rest_framework.pagination import PageNumberPagination
+from rest_framework.parsers import JSONParser
+from rest_framework.generics import GenericAPIView
+from rest_framework.views import APIView
+from rest_framework.response import Response
+from authors.models import Author 
+from authors.serializers import AuthorSerializer
+from authors.pagination import PaginationHandlerMixin
+
+class AuthorView(GenericAPIView):
+    pagination_class = PageNumberPagination
+
+    def get(self, request):
+        # PAGINATION WIP
+        # authors = Author.objects.all()
+        # print(authors)
+        # result_page = PageNumberPagination.paginate_queryset(authors, request)
+        # print(result_page)
+        # if result_page is not None:
+        #     serializer = AuthorSerializer(result_page, many=True)
+        #     return Response(self.get_paginated_response(serializer.data))
+        # return HttpResponse(status=404)
+
+        authors = Author.objects.all()
+        serializer = AuthorSerializer(authors, many=True)
+        return Response(serializer.data)
+    # POST request
+    # Fields from client payload are: displayName, github, profileImage
+    # Fields filled in by server: host, url, id
+    def post(self, request):
+        data = JSONParser().parse(request)
+        data = fillAuthorJSONPayload(request, data)
+        serializer = AuthorSerializer(data=data)
```

Comment:
I think you mentioned something about this in your comment to me, but this part doesn't work for me either, to make it work I removed line 33 and 34 and on line 35 I changed it very slightly so that it became
`serializer = AuthorSerializer(data=request.data)`

Reply:
I'll have to get this fixed, because we want to line 34 to get information from the request url